### ISSUE TITLE
add variable with weight decay

### ISSUE THREAD

**Issue by pribadihcr** [ppwwyyxx/tensorpack#484]
Hi @ppwwyyxx,

I want include weight decay in loss collection, as like the following example:
weight_decay = tf.multiply(tf.nn.l2_loss(var), 0.005, name='weight_loss')
tf.add_to_collection('losses', weight_decay)

How to define in tensorpack?
Thanks

**Comment by ppwwyyxx** [comment]
Any valid tensorflow code is valid in tensorpack. You can just use it the same way. Just remember that your model code might get called multiple times if trained with multiple GPUs.

tensorpack trainers don't use the loss collection for training. You can use it for your own purpose, of course.

**Comment by ppwwyyxx** [comment]
Feel free to reopen if you have more questions.
